x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/fulltext/Match.java

```diff
@@ -81,7 +84,8 @@ public class Match extends FullTextFunction implements Validatable {
         LONG,
         UNSIGNED_LONG,
         VERSION
-    );
+    ).filter(dataType -> dataType != SEMANTIC_TEXT || EsqlCapabilities.Cap.SEMANTIC_TEXT_TYPE.isEnabled()).collect(Collectors.toSet());
```

Comment:
I think this is more extensible if we use `DataType.UNDER_CONSTRUCTION` for the check, so this can be used in the future